### ISSUE TITLE
RazorGenerator uses `dynamic` ViewModel even when page class is non-generic

### ISSUE THREAD

**Issue by Jehoel** [RazorGenerator/RazorGenerator#109]
I use custom `WebViewPage`-subclasses in my projects.

My current project has a class-hierarchy deriving from `WebViewPage<TViewModel>`, like so:

    WebViewPage                            // standard
    WebViewPage<TViewModel>                      // standard
    WebViewPage2<TViewData,TViewModel> // in my own shared library
    FooPage<TViewData,TViewModel> // in the current website project, shared by all pages.
    BazPage : FooPage<BazViewData,BazViewModel> // concrete class used for a specific view

Most pages use `FooPage<TViewData,TViewModel>` in their `.cshtml` files and use the `@inherits` syntax (instead of `@model`) and it works fine, so `Qux.cshtml` file:

    @inherits FooPage<QuxViewData,QuxViewModel>
    @{ WriteLiteral("hello!"); }
    <div>foo</div>

generates this `Qux.cshtml.generated.cs`:

    namespace ASP {
        using /* ... */
        
        [GeneratedCodeAttribute("RazorGenerator", "2.0.0.0")]
        [PageVirtualPathAttribute("~Areas/Foo/Qux.cshtml");
        public partial class _Areas_Foo_Qux_cshtml : FooPage<QuxViewData,QuxViewModel> {
            /* ... */
        }
    }

However, if I want my `.cshtml` to derive from a concrete (non-generic) class, it doesn't work because (either RazorGenerator, or Razor itself) assumes it can use a `dynamic` generic-type argument. I never had this behaviour before when using Razor (without RazorGenerator) so I'm inclined to believe it's a bug in RazorGenerator.

This is the `Baz.cshtml` that causes the error:

    @inherits BazPage
    @{ WriteLiteral("hello!"); }
    <div>Baz</div>

it leads to this `Baz.cshtml.generated.cs` file:

    namespace ASP {
        using /* ... */
        
        [GeneratedCodeAttribute("RazorGenerator", "2.0.0.0")]
        [PageVirtualPathAttribute("~Areas/Foo/Baz.cshtml");
        public partial class _Areas_Foo_Baz_cshtml : BazPage<dynamic> {
            /* ... */
        }
    }

...which gives me this compiler error:

>     Severity	Code	Description	Project	File	Line
>    Error	CS0308	The non-generic type 'BazPage' cannot be used with type arguments

The system should see that the selected base class lacks generic type arguments and so should not provide or assume `<dynamic>` can be specified.

**Comment by Jehoel** [comment]
As a work-around, if I add a dummy type-argument to `BadPage` and then specify that in the `.cshtml` then it compiles:

BadPage.cs:

    class BadPage<TDummy> : ProjectPage<FooViewData,FooViewModel> {
        /* ... */
    }

Bad.cshtml:

    @inherits BadPage<Object>
    @{ WriteLiteral("hello!"); }
    <div>Baz</div>

**Comment by ronaldbarendse** [comment]
I agree this is a bug in RazorGenerator... Is there any progress in fixing this issue?
